Doppler.Sap/Mappers/BusinessPartner/BusinessPartnerMapper.cs

```diff
@@ -30,5 +33,69 @@ protected int MapGroupCode(DopplerUserDto dopplerUser)
 
             return groupCode;
         }
+
+        protected List<SapContactEmployee> GetContactEmployees(DopplerUserDto dopplerUser, string cardCode, string emailGroupCode)
+        {
+            var contactEmployees = (dopplerUser.BillingEmails != null && dopplerUser.BillingEmails[0] != String.Empty) ?
+                dopplerUser.BillingEmails
+                    .Select(x => new SapContactEmployee
+                    {
+                        Name = new MailAddress(x.ToLower()).User,
+                        E_Mail = x.ToLower(),
+                        CardCode = cardCode,
+                        Active = "tYES",
+                        EmailGroupCode = emailGroupCode
+                    })
+                    .Append(new SapContactEmployee
+                    {
+                        Name = new MailAddress(dopplerUser.Email.ToLower()).User,
+                        E_Mail = dopplerUser.Email.ToLower(),
+                        CardCode = cardCode,
+                        Active = "tYES",
+                        EmailGroupCode = emailGroupCode
+                    })
+                    .GroupBy(y => y.E_Mail)
+                    .Select(z => z.First())
+                    .ToList()
+                    :
+                    new List<SapContactEmployee>
+                    {
+                        new SapContactEmployee
+                        {
+                            Name = new MailAddress(dopplerUser.Email.ToLower()).User,
+                            E_Mail = dopplerUser.Email.ToLower(),
+                            CardCode = cardCode,
+                            Active = "tYES",
+                            EmailGroupCode = emailGroupCode
+                        }
+                    };
+
+            var sapContactEmployees = GetContactEmployeesWithoutRepeatedName(contactEmployees);
+            return sapContactEmployees;
+        }
+
+        private List<SapContactEmployee> GetContactEmployeesWithoutRepeatedName(List<SapContactEmployee> contacts)
+        {
+            var contactEmployeesGrouped = contacts.GroupBy(c => c.Name).Select(z => new { z.Key, Result = z.Select(c => c) });
```

Comment:
What do you think to add `.ToList()` ? because this variable to reduce positive false cases
`var contactEmployeesGrouped = contacts.GroupBy(c => c.Name).Select(z => new { z.Key, Result = z.Select(c => c) }).ToList();`

Reply:
Done